CMakeLists.txt

```diff
@@ -45,6 +45,25 @@ include(GNUInstallDirs)
 include(FeatureSummary)
 
 # https://medium.com/@alasher/colored-c-compiler-output-with-ninja-clang-gcc-10bfe7f2b949
+
+include(CMakePrintHelpers)
+
+# we transform the variables to CMAKE_PATH(slash based)
+file(TO_CMAKE_PATH "${CMAKE_INSTALL_FULL_BINDIR}" CMAKE_INSTALL_BINDIR_CMAKE_PATH)
+file(TO_CMAKE_PATH "${CMAKE_INSTALL_FULL_LIBDIR}" CMAKE_INSTALL_LIBDIR_CMAKE_PATH)
+file(TO_CMAKE_PATH "${CMAKE_INSTALL_FULL_DATADIR}" CMAKE_INSTALL_DATADIR_CMAKE_PATH)
+file(TO_CMAKE_PATH "${CMAKE_INSTALL_FULL_LOCALEDIR}" CMAKE_INSTALL_LOCALEDIR_CMAKE_PATH)
+get_filename_component(CMAKE_INSTALL_BINDIR_ABS "${CMAKE_INSTALL_FULL_BINDIR}" REALPATH BASE_DIR "${CMAKE_INSTALL_PREFIX}")
```

Comment:
Why this is needed? CMAKE_INSTALL_FULL_BINDIR should already be absolute. Do you want to resolve symlinks here? Is this because whereami library returns path with resolved symlinks? Also doesn't it make CMAKE_INSTALL_BINDIR_CMAKE_PATH variable unneeded?

Reply:
I don't remember that. The discussion in this PR from that time (bf51e038a0d501f0e1c88a6448e31a44cf75b27a) and my comment left in the code both are not very helpful:
`# we transform the variables to CMAKE_PATH(slash based)`

It might be important to transform on windows based systems, but this is just a guess? I would need to build dt on windows again to recall that. It also might be because of whereami. God, better comment of my own would be very helpful now.

CMAKE_INSTALL_BINDIR_CMAKE_PATH is not used, correct. Removed.

Reply:
CMAKE_INSTALL_<somename>_CMAKE_PATH variables removed. On linux no functionality regression noticeable. Might stem from the learning and debugging CMake and path resolution and trying out different solutions - time.

Reply:
Also updated the description to contain a list of possible startup scenarios. 
@parafin does that cover also all OsX scenarios?

Reply:
Regarding possible options to start dt on macOS - I think you covered them all (though some options are the same).

Now about these lines:

1. Converting using `TO_CMAKE_PATH` matters only on Windows, if I understand what it does correctly. Have you tested the current version of this PR (without these conversions) on Windows?
2. What is the purpose of the remaining `get_filename_component`? `CMAKE_INSTALL_FULL_BINDIR` is already absolute, isn't it? Is so, specifying `BASE_DIR` here makes no sense. The remaining effect of this line is to resolve symbolic links into a real path. That's a tricky scenario when bindir contains symbolic links, hence my question whether you know what whereami lib returns in this case? Maybe for safety we should run returned path through g_realpath ourselves, then we know for sure and this `get_filename_component` indeed makes sense (without `BASE_DIR` though probably). Everything will of course break if symbolic link is introduced in the bindir path after installation, but I suppose that's not a use-case we should care to support here.

Reply:
1. I am going to test this on windows again, soon. The last tests are 6 Months old.
2. I don't know what whereami returns if we are starting dt with a symbolic link. So I tried:
`ln -s ~/unstable/darktable_fix/cmake-run-dt-from-build-dir/bin/ ~/symdir
~/symdir/darktable -d dev
0.000106 application_directory: /home/david/unstable/darktable_fix/cmake-run-dt-from-build-dir/bin
0.000147 darktable.datadir: /home/david/unstable/darktable_fix/cmake-run-dt-from-build-dir/share/darktable
0.000170 darktable.plugindir: /home/david/unstable/darktable_fix/cmake-run-dt-from-build-dir/lib/darktable
0.000190 darktable.localedir: /home/david/unstable/darktable_fix/cmake-run-dt-from-build-dir/share/locale
0.000216 darktable.configdir: /home/david/.config/darktable
0.000232 darktable.cachedir: /home/david/.cache/darktable
0.000250 darktable.sharedir: /home/david/unstable/darktable_fix/cmake-run-dt-from-build-dir/share/darktable
0.000263 darktable.tmpdir: /tmp
0.000278 new_xdg_data_dirs: /home/david/unstable/darktable_fix/cmake-run-dt-from-build-dir/share/darktable:/home/david/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share:/var/lib/snapd/desktop
0.232634 kernel directory: /home/david/unstable/darktable_fix/cmake-run-dt-from-build-dir/share/darktable/kernels`
This also works, when removing the `BASE_DIR` as suggested.
Yes, we should not try to support use cases that don't cover my intention: Let the built binaries be starable without installing.

Reply:
OK, so it seems that whereami resolves symlinks too. Then everything is fine.